### PR TITLE
feat(button-toggle): describe ButtonToggle and ButtonToggleGroup components using design tokens (FE-4449)

### PR DESCRIPTION
**Pull request by mjaniczko-sage**
### Proposed behaviour

Describes the ButtonToggle and ButtonToggleGroup using design tokens. Removes all possible 'theme' in css files
Align icon colors with design
Updates tests after the above changes.

### Current behaviour
Component use the theme styled-component property.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/style/design-tokens/theme-compatibility.util.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ export default (theme) => ({
   colorsYin055: "rgba(0,0,0,0.55)",
   colorsYin065: "rgba(0,0,0,0.65)",
   colorsYin090: "rgba(0,0,0,0.9)",
+  colorsYang100: "rgba(255,255,255,1)",
```
Comment (jhajduk-sage) [inline src/style/design-tokens/theme-compatibility.util.js:24]: `theme-compatibility.util` is eventually removed. This has to be moved to `compatibility` section in theme-configs.

File: src/components/button-toggle-group/button-toggle-group.style.js [hunk 2]
```diff
@@ -13,22 +13,22 @@ const ButtonToggleGroupStyle = styled.div`
     `};
 
   ${StyledButtonToggleLabel} {
-    ${({ theme, info }) =>
+    ${({ info }) =>
```
Comment (mkrds) [inline src/components/button-toggle-group/button-toggle-group.style.js:16]: Get rid of `baseTheme` from this file, it's no longer needed

File: src/components/button-toggle/button-toggle.style.js [hunk 3]
```diff
@@ -53,36 +53,34 @@ const StyledButtonToggleLabel = styled.label`
   font-weight: 600;
   cursor: pointer;
 
-  ${({ theme }) => css`
-    border: 1px solid ${theme.colors.border};
-    background-color: ${theme.colors.white};
+  border: 1px solid var(--colorsActionMinor500);
```
Comment (mkrds) [inline src/components/button-toggle/button-toggle.style.js:56]: Same here, remove `baseTheme`

**Comment by mkrds** [comment]
Looks good to me, just fix the cypress tests and remove redundant `baseTheme` from style files

**Review by jhajduk-sage — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by mkrds — APPROVED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by Parsium — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**